vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectFactory.cs

```diff
@@ -112,6 +112,63 @@ protected override void CreateProject(string fileName, string location, string n
                 }
             }
 
+            // Solution properties
+            IVsSolution solution = this.Site.GetService(typeof(SVsSolution)) as IVsSolution;
+            Debug.Assert(solution != null, "Could not retrieve the solution service from the global service provider");
+
+            // We do not want to throw. If we cannot set the solution related constants we set them to empty string.
+            string solutionDirectory, solutionFile, userOptionsFile;
+            solution.GetSolutionInfo(out solutionDirectory, out solutionFile, out userOptionsFile);
+            if (solutionDirectory == null)
+            {
+                solutionDirectory = String.Empty;
+            }
+            if (solutionFile == null)
+            {
+                solutionFile = String.Empty;
+            }
+            string solutionFileName = (solutionFile.Length == 0) ? String.Empty : Path.GetFileName(solutionFile);
+            string solutionName = (solutionFile.Length == 0) ? String.Empty : Path.GetFileNameWithoutExtension(solutionFile);
+            string solutionExtension = String.Empty;
+            if (solutionFile.Length > 0 && Path.HasExtension(solutionFile))
+            {
+                solutionExtension = Path.GetExtension(solutionFile);
+            }
```

Comment:
`Path.GetExtension()` returns an empty string if the argument doesn't have an extension, including if it's emtpy; these 4 lines can be reduced to `var solutionExtension = Path.GetExtension(solutionFile);`.